### DISCUSSION TITLE
Kernel crashes after I change back to 2-core machine type

### DISCUSSION THREAD

**Discussion by misterskye** [earthlab-education/Earth-Analytics-2023-01-Intro#481] [Q&A06 Vector Data]
It looks like @eculler did not have this issue in the video, but I most certainly do have this problem.

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/47870472/c1c45860-5085-446f-9ca0-7fbe086bd370)

@eculler Is this a Codespaces issue?

**Comment by advythr** [comment]
I have this same error message, but for the code chunk that downloads data from the USGS Water Boundary Dataset. The error persists even when switching to the 4-core machine type.

The code chunk is:

`# Download the WBD
wbd_url = (
    "https://prd-tnm.s3.amazonaws.com"
    "/StagedProducts/Hydrography/WBD/National/GDB/WBD_National_GDB.zip")
wbd_dir = et.data.get_data(url=wbd_url)
wbd_dir`

Edit: I am using a GitHub Codespace and running on the _main_ kernel.

**Reply by eculler** [reply]
Hm, @advythr I'm not sure about that. Did you stop and restart your machine when you switched to 4 cores? I suppose you could try an 8-core machine for the download?

A more technical way to deal with this problem is to *stream* the file connection instead of using the earthpy get_data function. That way, the container doesn't have to hold the whole file in memory. I'm happy to walk you through how to do that in office hours, but unless more folks are having this problem I don't want to add an additional complex component to this assignment.

**Comment by eculler** [comment]
@misterskye I believe that means that you need to stick with the 4-core machine. If I recall, I found the same thing when I went through in the video.

**Comment by aholcomb23** [comment]
@misterskye Concerning the @eculler comment about the stop and restart of the codespace: I followed the video and changed the codespace to 2-core and stopped and restarted the codespace. I then switched back to the 4-core machine and stopped and restarted my codespace. When I tried to run my notebook again the code stopped at that same spot that your screenshot shows.

I ended up restarting my codespace, stopping the codespace and restarting again about 4 times before it started working.  Not sure if this is a codespace issue or what, but it did work after a few "reboot" processes. I hope the same solution works for you.

**Reply by misterskye** [reply]
Thanks for the info!  Yeah, I started and stopped several times, but to no avail.  I'm impressed with your tenacity, but by the second time, I was simply perplexed and frustrated ;-)  
I will give it a try.

**Reply by eculler** [reply]
Thanks @aholcomb23 - I hope we can find another solution because that does sound annoying.

What happens when you look at https://github.com/codespaces? Can you verify that the codespace you are using is listed there with the correct machine size? Please make sure to double check that the name in your url bar for the codespace matches the larger codespace.
